app/code/Magento/Checkout/Model/Cart.php

```diff
@@ -438,8 +438,14 @@ public function addProductsByIds($productIds)
                 }
                 $product = $this->_getProduct($productId);
                 if ($product->getId() && $product->isVisibleInCatalog()) {
+                    $request = null;
+                    $stockItem = $this->stockRegistry->getStockItem($productId, $product->getStore()->getWebsiteId());
+                    $minimumQty = $stockItem->getMinSaleQty();
+                    if ($minimumQty && $minimumQty > 0) {
+                        $request = $minimumQty;
```

Comment:
Complex logic related to inventory must not be added in `Checkout` module. We can have much more checks than MinSaleQty.

Reply:
@orlangur Are you check the `addProduct()` function of `\Magento\Checkout\Model\Cart` It also check the `MinSaleQty` with same logic. I took the reference from there.

Reply:
@mageprince does not look good as well, it should be better implemented as a plugin, probably legacy code.

Why do we need to duplicate this logic out of `addProduct` method?

Reply:
I think we misunderstood functions between addProduct() of `\Magento\Quote\Model\Quote` and `addProduct()` of `\Magento\Checkout\Model\Cart`.

**`addProduct()`** of `\Magento\Checkout\Model\Cart` **check the `minimumQty` of product** to addtocart
While **`addProductsByIds()`** of `\Magento\Checkout\Model\Cart` **not check the minimumQty** but it must required to solve this issue.

Reply:
@mageprince considering there is very similar check in [Cart.php](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Checkout/Model/Cart.php#L372-L380) I think it's better at least to move this check to private method and reuse in both functions. I think @orlangur would like this check to be moved to `CatalogInventory` module

Reply:
@orlangur @sivaschenko 
I updated the pull request as per your suggestion.
Please review it.

Reply:
@orlangur @sivaschenko any update on this pull request.

Reply:
@orlangur can you please review the changes.